### PR TITLE
Fix OIDC Custom Icons

### PR DESCRIPTION
**Pull request by deanhannigan**
## Description

Restored OIDC Icon fix - https://github.com/Budibase/budibase/pull/10978

Addresses: 
- https://github.com/Budibase/budibase/issues/10812

## Screenshots

Custom image displaying correctly as the selected option and in the option list
Screenshot 2023-06-20 at 18 14 06

Custom OIDC Icon displayed in the login page.
Screenshot 2023-06-20 at 18 14 33

### CODE REVIEW

File: packages/builder/src/pages/builder/auth/_components/OIDCButton.svelte [hunk 1]
```diff
@@ -29,9 +30,9 @@
     }
   })
 
-  $: src = !$oidc.logo
+  $: src = !oidcLogoCheck
     ? OidcLogo
-    : preDefinedIcons[$oidc.logo] || `/global/logos_oidc/${$oidc.logo}`
+    : preDefinedIcons[$oidc.logo] || oidcLogoCheck
```
Comment (melohagan) [inline packages/builder/src/pages/builder/auth/_components/OIDCButton.svelte:35]: The statement `preDefinedIcons[$oidc.logo] || oidcLogoCheck` will always be true here.

File: packages/worker/src/api/controllers/global/configs.ts [hunk 2]
```diff
@@ -301,13 +328,20 @@ export async function publicOidc(ctx: Ctx<void, GetPublicOIDCConfigResponse>) {
...
+      enrichOIDCLogos(oidcLogoConfig)
+    }
 
     if (!config) {
       ctx.body = []
     } else {
       ctx.body = [
         {
-          logo: config.logo,
+          logo: oidcLogoConfig
```
Comment (melohagan) [inline packages/worker/src/api/controllers/global/configs.ts:342]: Can this be simplified? Something like:

```js
logo: oidcLogoConfig?.config[config.logo] ?? config.logo
```

**Review by melohagan — CHANGES_REQUESTED** [review]
There's a boolean statement that needs double checked, unless I'm mistaken!

File: packages/worker/src/api/controllers/global/configs.ts (hunk 2 above)
Comment (aptkingston) [reply packages/worker/src/api/controllers/global/configs.ts:342]: Yep I agree with Mel here, definitely should be simplified into a one liner. On a separate note, the naming here is a little confusing haha.

File: packages/builder/src/pages/builder/auth/_components/OIDCButton.svelte [hunk 3]
```diff
@@ -29,9 +30,9 @@
     }
   })
 
-  $: src = !$oidc.logo
+  $: src = !oidcLogoCheck
```
Comment (aptkingston) [inline packages/builder/src/pages/builder/auth/_components/OIDCButton.svelte:33]: What was the purpose of pulling out `$oidc.logo` to a separate variable but then only using it once in the statement? Personally I think that if something can't be simplified to a one liner, then multi-line ternary statements should be broken out into functions to make it easier to understand the logic, and possible to maintain it in future.

File: packages/builder/src/pages/builder/auth/_components/OIDCButton.svelte (hunk 1 above)
Comment (deanhannigan) [reply packages/builder/src/pages/builder/auth/_components/OIDCButton.svelte]: I've refactored out the area in favour of `oidcLogoImageURL = preDefinedIcons[$oidc.logo] ?? $oidc.logo` which is largely the same.

`$oidc.logo` will either be one of the preset strings (`OidcLogo`,`Auth0` etc),  a custom logo file URL or `null`

If I chose the Auth0 icon in budibase, `$oidc.logo` would be set to 'Auth0' and `preDefinedIcons[$oidc.logo]` would  resolve to `/builder/assets/auth0-logo.png`

If the user selected one of their own images, `$oidc.logo` would resolve to something like `/files/signed/global/logos_oidc/mylogo.png`

In the event that absolutely nothing was set in `oidcLogoImageURL` after that , `OidcLogo` will be used as the default.

File: packages/worker/src/api/controllers/global/configs.ts [hunk 4]
```diff
@@ -280,13 +281,39 @@ export async function save(ctx: UserCtx<Config>) {
   }
 }
 
+function enrichOIDCLogos(oidcLogos: OIDCLogosConfig) {
+  if (!oidcLogos) {
+    return
+  }
+  oidcLogos.config = Object.keys(oidcLogos.config).reduce(
```
Comment (melohagan) [inline packages/worker/src/api/controllers/global/configs.ts:288]: Can we guarantee that `oidcLogos.config` will always be an object?

I recently fixed a null-pointer where I had to do `Object.keys(x || {})`

**Review by melohagan — APPROVED** [review]
LGTM!

File: packages/worker/src/api/controllers/global/configs.ts (hunk 4 above)
Comment (deanhannigan) [reply packages/worker/src/api/controllers/global/configs.ts]: It's unlikely to ever not be an object once its been created, but definitely better safe than sorry

**Pull request merged**